cursor.go

```diff
@@ -261,13 +261,27 @@ func (c *Cursor) Fetch(db SqlPreparer) (batch *RowBatch, paginationKeypos uint64
 		values:             batchData,
 		paginationKeyIndex: paginationKeyIndex,
 		table:              c.Table,
+		columns:            filterByTableColumns(columns, c.Table.Columns),
```

Comment:
Since this function is not used anywhere else, can we just have the logic inline here (to avoid having to jump around functions and remap the variable names) and also leave a comment above the block of code linking to either this PR or the original reporting issue? It's a bit of non-obvious code that will be hard to understand for people just reading it.

Reply:
It's kind of meaty to have in-line and the method interface helps to explain it a bit. It also doesn't have much to do with the bug itself, this is only for filtering out the virtual MD5 column we create for fingerprinting. But I can expand on that in a comment.